src/main/kotlin/com/terraformation/backend/search/SearchFilter.kt

```diff
@@ -12,15 +12,33 @@ enum class SearchFilterType {
 
 interface SearchNode {
   fun toCondition(): Condition
+
+  /**
+   * Converts this node and all its descendents from fuzzy to exact filtering. If this node does not
+   * currently contain any fuzzy filter criteria, this method _must_ return an object that tests
+   * equal to `this` (e.g., by just returning `this`).
+   */
+  fun toExactSearch(): SearchNode
+
   fun referencedSublists(): Set<SublistField>
 }
 
-class OrNode(private val children: List<SearchNode>) : SearchNode {
+data class OrNode(private val children: List<SearchNode>) : SearchNode {
```

Comment:
nice usage of data classes. Is a one of the benefits here that it provides equality checking?

Reply:
Yes, I did this to get the `equals` implementation.